### PR TITLE
Add statement for solving error in php 8.1

### PR DESCRIPTION
**Pull request by metroo**
When use php 8.1 to run program an error has been occurred to passing parameter because this structure was deprecated

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Fixed issues  |  #19684

### CODE REVIEW

**Comment by bizley** [comment]
Casting $value to string should be enough.

File: framework/db/conditions/LikeConditionBuilder.php [hunk 1]
```diff
@@ -78,6 +78,9 @@ public function build(ExpressionInterface $expression, array &$params = [])
             if ($value instanceof ExpressionInterface) {
                 $phName = $this->queryBuilder->buildExpression($value, $params);
             } else {
+                if(empty($value)){
```
Comment (lubosdz) [inline framework/db/conditions/LikeConditionBuilder.php:81]: This is BC break - e.g. will skip properly supplied `0`. 
Should check for non-empty string e.g. `trim($value) !== ""`.
Actually whole block should be optimized.

**Comment by metroo** [comment]
> Casting $value to string should be enough.
Since you cast $value to string ,when this variable is null the value was set to string with no length. that is a extra process to run in the next line when you should bind this string by params.

File: framework/db/conditions/LikeConditionBuilder.php (hunk 1 above)
Comment (metroo) [reply framework/db/conditions/LikeConditionBuilder.php:81]: when I use your solution , then receive this error

    "name": "PHP Deprecated Warning",
    "message": "trim(): Passing null to parameter #1 ($string) of type string is deprecated",

Comment (lubosdz) [reply framework/db/conditions/LikeConditionBuilder.php:81]: You are right, casting seems needed in any way, e.g. `trim((string)$value) !== ""`.
(how nicely it worked before 8.1)

Comment (rob006) [reply framework/db/conditions/LikeConditionBuilder.php:81]: @metroo Why would you have `null` in this scenario? `LikeCondition` accepts only string or array of strings. Casting `null` to empty string may hide some errors, as you should not really have `null` here.

Comment (lubosdz) [reply framework/db/conditions/LikeConditionBuilder.php:81]: @metroo Could you please show code example how do you get this error / reproduce error? 
[Seems to me](https://github.com/yiisoft/yii2/blob/57ae8aa995805950616589273488a6af2fd5746d/framework/db/conditions/LikeConditionBuilder.php#L61) that passing in non-empty array values could get through ..

Comment (metroo) [reply framework/db/conditions/LikeConditionBuilder.php:81]: when you set array with  one argument as null value
   $values = array(0=>null);

Comment (rob006) [reply framework/db/conditions/LikeConditionBuilder.php:81]: > when you set array with one argument as null value
> $values = array(0=>null);

And what is expected SQL query in that case? `field LIKE "%%"`?

**Comment by uaoleg** [comment]
I proposed another fix: https://github.com/yiisoft/yii2/pull/19828

**Pull request closed without merging**